### ISSUE TITLE
Fine-tune an open-source regulatory model and publish to HuggingFace

### ISSUE THREAD

**Issue by SdSarthak** [SdSarthak/AegisAI#92]
Fine-tune Mistral-7B-Instruct or Llama-3.2-3B on EU AI Act regulatory Q&A to replace the generic LLM default in the RAG chain.

**Steps:**
1. Use `backend/data/regulatory_qa.csv` as base training data
2. Apply QLoRA fine-tuning with PEFT on a free Colab A100
3. Benchmark against vanilla model on a held-out eval set
4. Publish LoRA adapter to HuggingFace Hub
5. Add `notebooks/finetune_regulatory_model.ipynb` Colab notebook
6. Update `.env.example` Ollama instructions for the published model

**Depends on:** regulatory Q&A CSV issue

**Comment by madsysharma** [comment]
Hi @SdSarthak , I would like to work on this as part of GSSoC 2026.

**Comment by Maha585** [comment]
Hello @SdSarthak 
I'm interested to fine tune this model and publish to HuggingFace.

**Comment by Kahby** [comment]
Hello @SdSarthak i would like to contribute to this as a part of GSSoc 26

**Comment by Kahby** [comment]
@SdSarthak why was i unassigned??

**Comment by SdSarthak** [comment]
Appears to glitching out, see the image
@Kahby I have assigned it, dont worry
<img width="938" height="90" alt="Image" src="https://github.com/user-attachments/assets/80f28116-5e58-4ca0-a530-2e94ebaf7285" />

**Comment by rsprajapati7** [comment]
hey @SdSarthak May I work on this issue?. I'm a data scientist with experience in NLP and fine-tuning, and I'm comfortable with QLoRA + PEFT workflows

**Comment by owwll** [comment]
Hi @SdSarthak! 👋

GSSoC 2026 contributor here (`@owwll`).

I’d like to work on this issue.

### My understanding of the task:

* Use `backend/data/regulatory_qa.csv` as the training dataset
* Fine-tune an open-source model such as Mistral-7B-Instruct or Llama-3.2-3B
* Apply QLoRA fine-tuning using PEFT on a Google Colab A100 GPU
* Evaluate the fine-tuned model against the base model on a held-out validation set
* Publish the resulting LoRA adapter to the Hugging Face Hub
* Add a reproducible Colab notebook at `notebooks/finetune_regulatory_model.ipynb`
* Update `.env.example` and Ollama instructions to show how to use the published model

### Expected Outcome:

* A domain-adapted regulatory model specialized for EU AI Act Q&A
* Improved answer quality compared to the default generic model
* A publicly available Hugging Face adapter that others can reuse
* Clear instructions for local deployment and inference

### Goal:

This enhancement will improve the accuracy and consistency of the RAG pipeline by using a model tuned specifically for regulatory and compliance questions.

I have experience with machine learning and LLM workflows, and this issue aligns strongly with my interest in AI and model fine-tuning.

I estimate it will take around 4–7 days to train, evaluate, publish, and document thoroughly.

Could you please assign this issue to me?

Thanks! 🙏

**Comment by Kahby** [comment]
hey @SdSarthak kindly take a look at the pr and tell me if something needs to be updated.

**Comment by Kahby** [comment]
Hey @SdSarthak , Since the GSSoC contribution phase officially starts on May 15th, will the GSSoC tracker count this PR
